### ISSUE TITLE
value too long for type character varying(16)

### ISSUE THREAD

**Issue by Whoops** [Yetangitu/owncloud-apps#7]
Using Owncloud 8 and Postgres I'm getting the exception below

    Exception: {"Message":"An exception occurred while executing 'INSERT INTO oc_opds_metadata (\"id\", \"updated\", \"date\", \"author\", \"title\", \"language\", \"publisher\", \"isbn\", \"copyright\", \"description\", \"subjects\", \"cover\", \"rescan\") VALUES (?,?,?,?,?,?,?,?,?,?,?,?,?)' with params [6487, \"2015-02-17 22:25:16\", \"\", \"{\\\"Paul Butcher\\\":\\\"Paul Butcher\\\"}\", \"Seven Concurrency Models in Seven Weeks (for Walton Hoops)\", \"en\", \"The Pragmatic Bookshelf, LLC (454466)\", \"978-1-937785-65-9\", \"Copyright \\u00a9 2014 The Pragmatic Programmers, LLC\", \"\", \"[\\\"Pragmatic Bookshelf\\\"]\", null, \"2015-02-24T22:25:17+00:00\"]:\n\nSQLSTATE[22001]: String data, right truncated: 7 ERROR: value too long for type character varying(16)","Code":0,"Trace":"#0 \/usr\/local\/www\/owncloud\/3rdparty\/doctrine\/dbal\/lib\/Doctrine\/DBAL\/DBALException.php(116): Doctrine\\DBAL\\Driver\\AbstractPostgreSQLDriver->convertException('An exception oc...', Object(Doctrine\\DBAL\\Driver\\PDOException))\n#1 \/usr\/local\/www\/owncloud\/3rdparty\/doctrine\/dbal\/lib\/Doctrine\/DBAL\/Statement.php(172): Doctrine\\DBAL\\DBALException::driverExceptionDuringQuery(Object(Doctrine\\DBAL\\Driver\\PDOPgSql\\Driver), Object(Doctrine\\DBAL\\Driver\\PDOException), 'INSERT INTO oc_...', Array)\n#2 \/usr\/local\/www\/owncloud\/lib\/private\/db\/statementwrapper.php(63): Doctrine\\DBAL\\Statement->execute(Array)\n#3 \/usr\/local\/www\/owncloud\/apps\/files_opds\/lib\/meta.php(123): OC_DB_StatementWrapper->execute(Array)\n#4 \/usr\/local\/www\/owncloud\/apps\/files_opds\/lib\/meta.php(190): OCA\\Files_Opds\\Meta::save(Array)\n#5 \/usr\/local\/www\/owncloud\/apps\/files_opds\/lib\/meta.php(139): OCA\\Files_Opds\\Meta::scan(6487)\n#6 \/usr\/local\/www\/owncloud\/apps\/files_opds\/lib\/files.php(36): OCA\\Files_Opds\\Meta::get(6487)\n#7 \/usr\/local\/www\/owncloud\/apps\/files_opds\/lib\/files.php(58): OCA\\Files_Opds\\Files::formatFileInfo(Object(OC\\Files\\FileInfo))\n#8 \/usr\/local\/www\/owncloud\/apps\/files_opds\/lib\/feed.php(61): OCA\\Files_Opds\\Files::formatFileInfos(Array)\n#9 \/usr\/local\/www\/owncloud\/apps\/files_opds\/index.php(68): OCA\\Files_Opds\\Feed::serveFeed('\/Books\/programm...', '6274')\n#10 \/usr\/local\/www\/owncloud\/lib\/private\/route\/route.php(135) : runtime-created function(1): require_once('\/usr\/local\/www\/...')\n#11 [internal function]: __lambda_func(Array)\n#12 \/usr\/local\/www\/owncloud\/lib\/private\/route\/router.php(250): call_user_func('\\x00lambda_10681', Array)\n#13 \/usr\/local\/www\/owncloud\/lib\/base.php(763): OC\\Route\\Router->match('\/apps\/files_opd...')\n#14 \/usr\/local\/www\/owncloud\/index.php(36): OC::handleRequest()\n#15 {main}","File":"\/usr\/local\/www\/owncloud\/3rdparty\/doctrine\/dbal\/lib\/Doctrine\/DBAL\/Driver\/AbstractPostgreSQLDriver.php","Line":91} 

Note that I believe MySQL will quietly truncate in the same situation so there is likely data loss happening there.

**Comment by Yetangitu** [comment]
Hmmm, that seems to be the ISBN field, and that ISBN is longer than 16 characters due to the extraneous dash. I guess I better filter out all non-relevant data (ie anything except [0-9xX] before the ISBN value is stored.

Fix committed (https://github.com/Yetangitu/owncloud-apps/commit/9a03a36c8f8224db944e25eada41922156b395e8), you can either patch it in place or pull it in. Since PostgreSQL throws an error instead of truncating the value you should be allright when it comes to data loss.

Those who use MySQL or Sqlite could either schedule a rescan or trigger one for only affected titles by running the following query:

MySQL:
```sql
update oc_opds_metadata set rescan=now() where char_length(isbn) > 10 and (char_length(isbn) - char_length(replace(isbn,"-",""))) > 3;
```

Sqlite:
```sql
update oc_opds_metadata set rescan=datetime("now") where length(isbn) > 10 and (length(isbn) - length(replace(isbn,"-",""))) > 3;
```

Thanks for the bug report!

**Comment by Yetangitu** [comment]
Fixed in 0.6.5
